src/adagrams.js

```diff
@@ -1,15 +1,111 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const LETTER_POOL = {
+    'A': 9,
+    'B': 2, 
+    'C': 2,
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9,
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2,  
+    'Q': 1, 
+    'R': 6,
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+  };
+  
+  let avaliableLetters = [];
+  for (const [letter, freq] of Object.entries(LETTER_POOL)){
+  let i = 0; i < freq; i++
+  avaliableLetters.push(letter);
+  }
+  let hand = [];
+  for (let i = 0; i < 10; i ++) {
+  let randomNum = Math.floor(Math.random() * avaliableLetters.length);
+  hand.push(avaliableLetters[randomNum]);
+  avaliableLetters.splice(randomNum, 1);
+}
+return hand;
+}; 
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  for (let i = 0; i < input.length; i ++){
+    console.log(input[i])
+    if (!lettersInHand.includes(input[i])){
+      return false; 
+    } else if (lettersInHand.includes(input[i])) {
+      lettersInHand.splice(input[i], 1);
+      console.log(lettersInHand)
+      console.log(input)
+    }
+    } 
+  return true; 
+  } 
+
+function getKeyByValue (obj, val) {
+    return Object.entries(obj).find(i => i[1] === val);
+}
 
 export const scoreWord = (word) => {
```

Comment:
⚠️ This works, but it seems a little awkward.

Looping through they keys to find a key that has a specific value seems to be doing the opposite of what you should be doing with a JS object.  Maybe instead make the array of letters be the key and the value be the score?  You can actually do this I think, either with an array or using a Set as the key.

You could also make a bigger dictionary with individual letters being the key and the value being the score of that letter something like:

```javascript 
const scores = {
  'A': 1,
  'B': 3,
...
};
```